ads/models.py

```diff
@@ -1,3 +1,67 @@
 from django.db import models
+from django.utils.text import slugify
+from django.urls import reverse
+from taggit.managers import TaggableManager
+from django.core.exceptions import ValidationError
 
-# Create your models here.
+
+class Category(models.Model):
+    name = models.CharField(max_length=255, unique=True, blank=False)  
+    slug = models.SlugField(max_length=200, unique=True, blank=True)
+    description = models.TextField(blank=True) 
+    created_at = models.DateTimeField(auto_now_add=True)  
+
+    def __str__(self):
+        return self.name  
+    
+    def save(self, *args, **kwargs):
+        if not self.slug:
+            self.slug = slugify(self.name)
+        super().save(*args, **kwargs)
+
+
+class Ads(models.Model):
+    title = models.CharField(max_length=255, blank=False, null=False)
+    category = models.ForeignKey(Category, on_delete=models.CASCADE)
+    slug = models.SlugField(max_length=255, unique=True, blank=True)
+    image = models.ImageField(upload_to='ads/%Y/%m/%d/', blank=False, null=False)
+    description = models.TextField(blank=False, null=False)
+    tags = TaggableManager()  # Using TaggableManager for tags
+    location = models.CharField(max_length=255, blank=False, null=False)
+    postal_code = models.CharField(max_length=20, blank=False, null=False)
+    contact_info = models.CharField(max_length=255, blank=False, null=False)
+    price = models.DecimalField(max_digits=10, decimal_places=2, blank=False, null=False)
+    show_contact_info = models.BooleanField(default=True)
+    event_start_date = models.DateTimeField(null=True, blank=True)
+    event_end_date = models.DateTimeField(null=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
```

Comment:
There is no user field in the ad ? Who posted this ad ?

Reply:
i will add the user field sir.